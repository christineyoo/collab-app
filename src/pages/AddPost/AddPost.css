* {
    box-sizing: border-box;
  }
  
  .add-post {
    text-align: left;
    margin: 5%;
  }
  
  header {
    padding: 12% 0 0 0;
  }

  input, textarea, select {
    padding: 1rem;
    font-size: 1rem;
  }

  .field {
    margin: 0 0 0.5rem 0;
  }

  textarea {
    max-width: 95%;
  }