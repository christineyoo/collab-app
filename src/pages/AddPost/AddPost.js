import React, { Component } from 'react';
import UserNav from '../../Organisms/Nav/UserNav.js';
import Footer from '../../Organisms/Footer/Footer.js';
import './AddPost.css';

class AddPost extends Component {
  render() {
    return (
      <>
        <UserNav />
        <main className='add-post'>
          <header className='header'>
            <h1>Add a new post.</h1>
            <form>
              <label for='title'>Title</label>&nbsp;
              <input id='title' type='text' name='title' required />
              <br />
              <label for='content'>Content</label>
              <br />
              <textarea
                id='content'
                name='content'
                rows='4'
                cols='50'
              ></textarea>
              <br />
              <label for='groups'>Group</label>&nbsp;
              <select name='group' id='group'>
                <option value='algebra-1'>Algebra 1</option>
                <option value='geometry'>Geometry</option>
                <option value='algebra-2'>Algebra 2</option>
                <option value='ids'>Intro to Data Science</option>
                <option value='math'>Math department</option>
              </select>
              <br />
              <button type='submit'>Add Post</button>
            </form>
          </header>
        </main>
        <Footer />
      </>
    );
  }
}

export default AddPost;
