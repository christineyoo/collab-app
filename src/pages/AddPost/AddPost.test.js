import React from 'react';
import renderer from 'react-test-renderer';
import AddPost from './AddPost';

it('renders AddPost UI as expected', () => {
    const tree = renderer.create(<AddPost />).toJSON();
    expect(tree).toMatchSnapshot();
})