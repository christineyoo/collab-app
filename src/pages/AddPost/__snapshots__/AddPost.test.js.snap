// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders AddPost UI as expected 1`] = `
Array [
  <nav>
    <a
      exact={true}
      href="/"
      id="nav"
      onClick={[Function]}
    >
      Collab
    </a>
  </nav>,
  <div
    id="add-post-bg"
  >
    <a
      href="/all-posts"
      onClick={[Function]}
    >
      <span
        id="add-post-button"
      >
        Go Back
      </span>
    </a>
    <div
      className="add-post"
    >
      <header
        className="header"
      >
        <h1
          id="add-post-h1"
        >
          Add a new post
        </h1>
        <br />
        <form
          onSubmit={[Function]}
        >
          <div
            className="field"
          >
            <label
              htmlFor="title"
            >
              Title
            </label>
            <br />
            <input
              id="title"
              name="title"
              onChange={[Function]}
              required={true}
              type="text"
            />
          </div>
          <br />
          <div
            className="field"
          >
            <label
              htmlFor="content"
            >
              Content
            </label>
            <br />
            <textarea
              cols="100"
              id="content"
              name="content"
              onChange={[Function]}
              rows="15"
            />
          </div>
          <br />
          <div
            className="field"
          >
            <label
              htmlFor="groups"
            >
              Group
            </label>
            <br />
            <select
              id="group"
              name="group"
              onChange={[Function]}
            >
              <option
                value={1}
              >
                Algebra 1
              </option>
              <option
                value={2}
              >
                Geometry
              </option>
              <option
                value={3}
              >
                Algebra 2
              </option>
              <option
                value={4}
              >
                Intro to Data Science
              </option>
            </select>
          </div>
          <br />
          <div
            className="field"
          >
            <label
              htmlFor="names"
            >
              Author
            </label>
            <br />
            <input
              id="name"
              name="name"
              onChange={[Function]}
              required={true}
              type="text"
            />
          </div>
          <br />
          <button
            id="add-post-button"
            type="submit"
          >
            Add Post
          </button>
        </form>
      </header>
    </div>
  </div>,
  <footer
    className="footer-style"
  >
    <div
      className="container"
    >
      <h4
        id="footer-h4-hh"
      >
        Copyright © 2021 Christine Yoo
      </h4>
      <ul
        id="footer-menu"
      >
        <li>
          <i
            className="fab fa-linkedin-in"
          />
          <a
            href="https://www.linkedin.com/in/christine-yoo-cy/"
            rel="noreferrer"
            target="_blank"
          >
             LinkedIn
          </a>
        </li>
        <li>
          <i
            className="fab fa-github"
          />
          <a
            href="https://github.com/christineyoo"
            rel="noreferrer"
            target="_blank"
          >
             GitHub
          </a>
        </li>
      </ul>
    </div>
  </footer>,
]
`;
